libtransmission/peer-mgr.cc

```diff
@@ -244,15 +288,28 @@ class tr_swarm
         TR_ASSERT(stats.peer_count == peerCount());
 
         delete peer;
+
+        if (was_incoming)
+        {
+            [[maybe_unused]] auto const port_empty = std::empty(peer_info->listen_port());
+            if (incoming_pool.erase(socket_address) != 0U)
+            {
+                TR_ASSERT(port_empty);
+            }
+        }
+        graveyard_pool.erase(socket_address);
     }
 
-    void removeAllPeers()
+    void remove_peer(tr_peerMsgs* peer)
```

Comment:
_(minor, opinion)_ Please move this to right above `remove_peer(iter)` since this is a one-liner that calls it; that way the readability flow from `remove_peer(peer)` to `remove_peer(iter)` is very readable

Reply:
36015c0